demo-app/src/data_generation/election_data.rs

```diff
@@ -0,0 +1,234 @@
+use super::*;
+
+struct CallGenerator {
+    election_admin_nonce: u64,
+    election_admin: MockPublicKey,
+}
+
+impl CallGenerator {
+    fn new() -> Self {
+        Self {
+            election_admin_nonce: 0,
+            election_admin: MockPublicKey::try_from("election_admin").unwrap(),
+        }
+    }
+
+    fn create_voters_and_vote(
+        &mut self,
+    ) -> Vec<(MockPublicKey, election::call::CallMessage<MockContext>, u64)> {
+        let mut messages = Vec::default();
+
+        let set_candidates_message = election::call::CallMessage::SetCandidates {
+            names: vec!["candidate_1".to_owned(), "candidate_2".to_owned()],
+        };
+
+        messages.push((
+            self.election_admin.clone(),
+            set_candidates_message,
+            self.election_admin_nonce,
+        ));
+        self.election_admin_nonce += 1;
```

Comment:
Non-blocking: Should we use a method to get and increment the nonce so that we don't have to remember the `+=1` every time?

Reply:
Done